sphinxcontrib/jupyter/writers/translate_all.py

```diff
@@ -825,7 +825,7 @@ def split_uri_id(cls, uri):
     @classmethod
     def add_extension_to_inline_link(cls, uri, ext):
         if "." not in uri:
-            if uri[0] == "#":
+            if len(uri) > 0 and uri[0] == "#":
```

Comment:
Isn't `uri[0] == "#"` still going to be evaluated in this `and` condition? I think that will still cause a traceback?

Reply:
So it is left to right! -- neat.

Reply:
Yeah python short-circuits the `and` and `or` conditions